packages/playwright-test/src/reporters/junit.ts

```diff
@@ -133,6 +133,94 @@ class JUnitReporter implements Reporter {
     };
     entries.push(entry);
 
+    // Xray Test Management supports testcase level properties, where additional metadata may be provided
+    // some annotations are encoded as value attributes, other as cdata content
+    const properties: XMLEntry = {
+      name: 'properties',
+      children: [] as XMLEntry[]
+    };
+
+    const textValueBasedAnnotations = ['test_id', 'test_key', 'test_summary', 'requirements'];
+    for (const annotationType of textValueBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType,
+            value: (annotation?.description ? annotation.description : '')
+          }
+        };
+        properties.children?.push(property);
+      }
+    }
+
+    const textContentBasedAnnotations = ['test_description'];
+    for (const annotationType of textContentBasedAnnotations) {
+      const annotation = test.annotations.find(annotation => {
+        return annotation.type === annotationType;
+      });
+      if (annotation !== undefined) {
+        const property: XMLEntry = {
+          name: 'property',
+          attributes: {
+            name: annotationType
+          },
+          text: annotation.description
```

Comment:
Why do we have to put this one as text, and not as `value=` as others? I'd assume that these are interchangeable, it's just a matter of escaping.

Reply:
For the time being, Xray processes some properties from a `value` attribute and other ones from inner text, due to their nature. It's not interchangeable.
So we have a set of well-known properties are are either mapped as "myproperty1" or as shown for "myproperty2".

`
<property name="myproperty1" value="dummyvalue"/>
<property name="myproperty2">
 a very long description .....
</property>
`

To make this mechanism generic and process either from the value or from the inner text,  we would need to change Xray. In XML there are limitations for the usage of the `value` attribute, as AFAIK though.

One option here moving forward, in order to make it more agnostic and not have any hardcoding of the annotations that should be either mapped to a value or to inner text, would be to have some sort of explicitly differentiate that whenever using the annotations. However, in playwright we just have the `type` and the `description` fields, correct? 

`
testInfo.annotations.push({ type: 'test_id', description: '1234' });
`

I don't see here a way of signalling it from the outside.. unless we would use a prefix on the `type` or on the  `description` or something like that.
Any ideas?

Reply:
I see, thank you for clarifying. Let's go with options for now, and provide a nice snippet in the docs:

```js
// JUnit reporter config for XRay.
const xrayOptions = {
  // Whether to add <properties> with all annotations.
  embedAnnotationsAsProperties: true,

  // By default, annotation is reported as <property name='' value=''>.
  // These annotations are reported as <property name=''>value</property>.
  textContentAnnotations: ['test_description'],

  // This will create a "testrun_evidence" property that contains all attachments.
  // Disables [[ATTACHMENT|path]] in the <system-out>.
  embedAttachmentsAsProperty: 'testrun_evidence',

  // Where to put the report.
  outputFile: './xray-report.xml',
};

module.exports = {
  reporter: [['list'],  ['junit', xrayOptions]]
};
```

Reply:
done, updated the PR, including the docs